### PR TITLE
Mmo rust

### PR DESCRIPTION
**Pull request by Tarnadas**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,8 +21,6 @@ COPY proto/ ./proto
 RUN npm run build
 
 COPY src/favicon.ico ./dist/
-COPY src/mmo/assets/ ./dist/mmo/assets
```
Comment (Tarnadas) [inline Dockerfile:24]: this has been moved to Webpack via CopyWebpackPlugin

File: src/mmo/socket.js [hunk 2]
```diff
@@ -94,34 +94,6 @@ const measureLatency = (ping_proto) => {
...
-        accessCodeMsg.setAccessCode(url.searchParams.get('code'))
-        accessCodeMsg.setType(state.type)
-        const initializationMsg = new InitializationMsg()
-        initializationMsg.setAccessCodeMsg(accessCodeMsg)
-        const sm64jsMsg = new Sm64JsMsg()
-        sm64jsMsg.setInitializationMsg(initializationMsg)
-        const rootMsg = new RootMsg()
-        rootMsg.setUncompressedSm64jsMsg(sm64jsMsg)
-        sendData(rootMsg.serializeBinary())
-    } else if (process.env.PRODUCTION == 0) {
```
Comment (Tarnadas) [inline src/mmo/socket.js]: should I be adding an easier way for local development? I deleted this, but haven't added it to the REST method

File: src/mmo/socket.js [hunk 3]
```diff
@@ -452,15 +422,17 @@ export const sendChat = ({ message }) => {
     sendData(rootMsg.serializeBinary())
 }
 
-const redirect_uri = encodeURIComponent(url.protocol == "https:" ? 'https://sm64js.com' : 'http://localhost:9300')
+const redirect_uri = encodeURIComponent(process.env.NODE_ENV === 'rust'
+    ? `${url.protocol}//${window.location.host}`
+    : url.protocol == "https:" ? 'https://sm64js.com' : 'http://localhost:9300')
 
-const discord_client_id = process.env.DISCORD_CLIENT_ID
+const discord_client_id = "807123464414429184"
```
Comment (Tarnadas) [inline src/mmo/socket.js:441]: ClientIDs are no secrets and they will be visible to any user that downloads the website's assets, so there is no need to store them in an env variable. Only the server has secrets

Comment (Tarnadas) [reply src/mmo/socket.js:441]: you will also have to override these with your clientIDs

File: Dockerfile (hunk 1 above)
Comment (sm64js) [reply Dockerfile:24]: cool

File: src/mmo/socket.js (hunk 2 above)
Comment (sm64js) [reply src/mmo/socket.js]: yea, we'll want to make sure there is a way to test MMO locally without having to setup a bunch of Oauth or other prod stuff

File: src/mmo/socket.js (hunk 3 above)
Comment (sm64js) [reply src/mmo/socket.js:441]: I realize clientIDs are no secret... I just thought it was a good idea to have as environment variable to make it easier to change if that was ever needed...

Or if someone wanted to setup their own server, they could use their own clientID

Comment (sm64js) [reply src/mmo/socket.js:441]: override with your clientIDs? what do you mean?

Comment (Tarnadas) [reply src/mmo/socket.js:441]: ok gotcha, I will add it back.
I meant that you replace my clientID with yours, but I will add back the env variable

**Pull request merged**